openedx/openedx-event-sink-clickhouse
# Add Course Published event listener and plugin plumbing

PR is ready for review. This has worked for me in a Tutor nightly build using the accompanying branch of the OARS plugin: https://github.com/openedx/tutor-contrib-oars/pull/35

Testing: 

- (in an otherwise working OARS env)
- Install the OARS plugin branch above
- `tutor config save` (to get the ClickHouse config)
- `tutor images build openedx --no-cache` (to get the plugin installed)
- `tutor local do init -l oars` (to get the new database and tables created)
- `tutor local start`
- Make sure you have the demo course loaded
- Edit a text block in the course
- You should see course data in ClickHouse's `event_sink.course_blocks` and `event_sink.course_relationships`
- If there are errors they will likely show up in the cms-worker container

Known issue: The data is not versioned, I will need to update both tables to have a unique id of some variety so we can tell which versions go with which events.

**Merge checklist:**
Check off if complete *or* not applicable:
